src/util-logopenfile.c

```diff
@@ -121,6 +121,63 @@ static int SCLogUnixSocketReconnect(LogFileCtx *log_ctx)
     return log_ctx->fp ? 1 : 0;
 }
 
+static int SCLogFileWriteSocket(const char *buffer, int buffer_len,
+        LogFileCtx *ctx)
+{
+    int tries = 0;
+    int ret = 0;
+    bool reopen = false;
+    int flags = 0;
+
+    if (ctx->fp == NULL && ctx->is_sock) {
+        SCLogUnixSocketReconnect(ctx);
+    }
+
+    /* Set flags for a non-blocking write if live. */
+    if (TimeModeIsLive()) {
```

Comment:
I would like to see this handled globally somehow. Seems wasteful to check this on every write. It won't ever change at runtime.